### PR TITLE
MVP Fertility- part 1

### PR DESCRIPTION
**Pull request by LoraArmstrong**
- Add test inputs for new fertility functions
- Add R outputs to test new fertility functions against
- Add schemas + coercion for newly added data
- Project ASFR (final year historic fixed) fn + tests
- Combine births by sex fn + tests
- Note- ideally a `gather` function should be added to witan.datasets. For now there is a custom version here to use for combining births by sex only.

### CODE REVIEW

File: src/witan/models/dem/ccm/fert/fertility_mvp.clj [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                     (ds/add-column :sex (repeat "F"))
+                     (ds/rename-columns {:f :births})
+                     (ds/select-columns [:gss-code :sex :births]))
+        births-m (-> births-by-mf
+                     (ds/add-column :sex (repeat "M"))
+                     (ds/rename-columns {:m :births})
+                     (ds/select-columns [:gss-code :sex :births]))]
+    (ds/join-rows births-m births-f)))
+
+(defn combine-into-births-by-sex
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/fert/fertility_mvp.clj]: change to workflowfn

File: src/witan/models/dem/ccm/schemas.clj [hunk 2]
```diff
@@ -109,4 +109,18 @@
...
+  (make-ordered-ds-schema [[:gss-code s/Str] [:sex s/Str] [:age s/Int]
+                           [:year s/Int] [:fert-rate double]]))
+
+(def ProjFixedASFRSchema
+  (make-ordered-ds-schema [[:gss-code s/Str] [:sex s/Str] [:age s/Int]
+                           [:fert-rate double]]))
+(def BirthsAgeSexMotherSchema
+  (make-ordered-ds-schema [[:gss-code s/Str] [:sex s/Str] [:age s/Int]
+                           [:year s/Int] [:popn double]
+                           [:fert-rate double] [:births double]]))
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/schemas.clj]: remove fert rate from handmade input and schema

File: test/witan/models/dem/ccm/fert/fertility_mvp_test.clj [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+;;                             fn from R in fertility module, to use
+;;                             until a Clojure version exists
+
+(def fertility-inputs (ld/load-datasets
+                       {:historic-asfr
+                        "resources/test_data/fertility/bristol_historic_asfr.csv"
+                        :births-age-sex-mother
+                        "resources/test_data/fertility/bristol_proj_births_age_sex_mother_2015.csv"}))
+
+(def params {:fert-last-yr 2014 :pm (/ 105 205)})
```
Comment (sunnytownsend) [inline test/witan/models/dem/ccm/fert/fertility_mvp_test.clj]: add double around division

File: src/witan/models/dem/ccm/fert/fertility_mvp.clj [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  "Takes a dataset with births by age and sex of mother, and pm, the fraction
+   of births that are male (from 0-1)"
+  [births-by-agesex pm]
+  (-> births-by-agesex
+      (wds/rollup :sum :births [:gss-code])
+      (wds/add-derived-column :m [:births] (fn [b] (* pm b)))
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/fert/fertility_mvp.clj]: add double around multiplication

File: src/witan/models/dem/ccm/fert/fertility_mvp.clj [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      (wds/add-derived-column :f [:births] (fn [b] (* (- 1 pm) b)))
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/fert/fertility_mvp.clj]: add double around multiplication

File: test/witan/models/dem/ccm/fert/fertility_mvp_test.clj [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(def fertility-outputs-r (ld/load-datasets
+                          {:projected-asfr-finalyrfixed
+                           "resources/test_data/fertility/bristol_initial_projected_fertility_rates_2015.csv"
+                           :births-by-sex
+                           "resources/test_data/fertility/bristol_proj_births_by_sex_2015.csv"}))
+
+(deftest project-asfr-finalyrhist-fixed-test
+  (testing "Fertiliy rates are projected correctly."
```
Comment (sunnytownsend) [inline test/witan/models/dem/ccm/fert/fertility_mvp_test.clj]: missing t

**Comment by sunnytownsend** [comment]
lgtm :shipit: once conflicts and comments addressed

**Comment by tcoupland** [comment]
:shipit: can see the need for that gather function though!

**Pull request merged**